server/src/main/java/org/eclipse/openvsx/publish/PublishExtensionVersionHandler.java

```diff
@@ -179,4 +204,17 @@ public void publishAsync(FileResource download, Path extensionFile, ExtensionSer
         // Update whether extension is active, the search index and evict cache
         service.activateExtension(extVersion, extensionService);
     }
+
+    public void mirror(FileResource download, Path extensionFile) {
+        var extVersion = download.getExtension();
+        service.mirrorResource(download);
+        try(var processor = new ExtensionProcessor(extensionFile)) {
+            Consumer<FileResource> consumer = resource -> {
+                service.mirrorResource(resource);
+            };
+
+            processor.getFileResources(extVersion).forEach(consumer);
+            // don't store file resources, they can be generated on the flight to avoid traversing entire zip file
```

Comment:
What do you mean? `mirrorResource` still persists the FileResource in the database.

Reply:
It is to improve mirroring performance. There is a shortcut: https://github.com/eclipse/openvsx/pull/586#discussion_r1033765662